### DISCUSSION TITLE
define sling asset using source and target connections

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17907] [Q&A] [answered]
Hi, I need a suggestion w.r.t embedded elt concept in dagster. using the yaml specs(source and target connections) defined by the user, Is there a way we define/generate sling asset by making the connection like this:-

`sling = SlingResource(source_connection=source, target_connection=target)`

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1699639485571539?thread_ts=1699639485.571539&cid=C01U954MEER).*

**Comment by PedramNavid** [comment, marked as answer]
Hi, I think I need more context. What does `source` and `target` refer to here? 

```
sling_resource = SlingResource(
    source_connection=SlingSourceConnection(type="file"),
    target_connection=SlingTargetConnection(
        type="sqlite", connection_string="sqlite://path/to/sqlite.db"
    ),
)
```

You can certainly use a source/target connection like above.
